### PR TITLE
Client files to active storage

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/controllers/clients/files_controller.rb [hunk 1]
```diff
@@ -46,32 +46,27 @@ def new
     def create
       @file = file_source.new
       @file.errors.add :tag_list, 'You must specify file contents' if file_params[:tag_list].blank?
-      @file.errors.add :file, 'No uploaded file found' unless file_params[:file]
+      @file.errors.add :client_file, 'No uploaded file found' unless file_params[:client_file]
```
Comment (awisema) [inline app/controllers/clients/files_controller.rb]: Philosophical question -- should we add parens to method calls when we find them like this?
```suggestion
      @file.errors.add(:client_file, 'No uploaded file found') unless file_params[:client_file]
```
And, do/should we have a style convention for it? When there are multiple params like this, I like to see the parens -- but in the `render :new` below, I think `render(:new)` looks *less* readable to me...

File: app/controllers/clients/files_controller.rb [hunk 2]
```diff
@@ -46,32 +46,27 @@ def new
...
+          allowed_params.merge(
+            file: 'See S3', # Temporary until we remove the column
+            client_id: @client.id,
+            user_id: current_user.id,
+            visible_in_window: window_visible?(allowed_params[:visible_in_window]),
+            consent_form_confirmed: allowed_params[:consent_form_confirmed] || GrdaWarehouse::Config.get(:auto_confirm_consent),
+            coc_codes: allowed_params[:coc_codes].reject(&:blank?),
+          ),
+        )
+        @file.name = "#{@file.client_file.filename.base}#{@file.client_file.filename.extension_with_delimiter}"
```
Comment (awisema) [inline app/controllers/clients/files_controller.rb]: Pull this out as a method?

File: app/models/grda_warehouse/client_file.rb [hunk 3]
```diff
@@ -331,23 +330,29 @@ def self.available_tags
...
     def as_preview
-      return content unless content_type == 'image/jpeg'
+      return client_file.download unless client_file.variable?
 
-      image = MiniMagick::Image.read(content)
-      image.auto_level
-      image.strip
-      image.resize('1920x1080')
-      image.to_blob
+      client_file.variant(resize_to_limit: [400, 400]).processed.download
```
Comment (awisema) [inline app/models/grda_warehouse/client_file.rb]: I think this and `as_thumb` below have their sizes swapped?

**Review by awisema — CHANGES_REQUESTED** [review]

File: app/models/grda_warehouse/client_file.rb (hunk 3 above)
Comment (eanders) [reply app/models/grda_warehouse/client_file.rb]: yep!

File: app/controllers/clients/files_controller.rb (hunk 1 above)
Comment (eanders) [reply app/controllers/clients/files_controller.rb]: I agree.  I haven't found the cop for it yet.  Fixed these though.

File: app/controllers/clients/files_controller.rb (hunk 2 above)
Comment (eanders) [reply app/controllers/clients/files_controller.rb]: Good call

**Review by awisema — APPROVED** [review]

**Pull request merged**